useragent/useragent.go

```diff
@@ -49,34 +51,43 @@ func GetUserAgent() string {
 }
 
 func getUserAgentBaseAndVersion() string {
+	Logger.Trace("Attempting to obtain user agent and version")
 	return fmt.Sprintf("%s/%s", CLIENTTOOL, buildversion.BuildVersion)
 }
 
 func checkCIEnvironments(callTree string) string {
 	if checkForCISystem("CIRCLECI") {
+		Logger.Trace("CircleCI usage")
 		return getUserAgent("circleci", callTree)
 	}
 	if checkForCISystem("BITBUCKET_BUILD_NUMBER") {
+		Logger.Trace("BitBucket usage")
 		return getUserAgent("bitbucket", callTree)
 	}
 	if checkForCISystem("TRAVIS") {
+		Logger.Trace("TravisCI usage")
 		return getUserAgent("travis-ci", callTree)
 	}
 	if checkForCISystem("GITLAB_CI") {
+		Logger.Trace("GitLab usage")
 		return getUserAgent("gitlab-ci", callTree)
 	}
 	if checkIfJenkins() {
+		Logger.Trace("Jenkins usage")
 		return getUserAgent("jenkins", callTree)
 	}
 	if checkIfGitHub() {
 		id := getGitHubActionID()
+		Logger.WithField("gh_action_id", id).Trace("GitHub Actions usage")
 		return getUserAgent(fmt.Sprintf("github-action %s", id), callTree)
 	}
 
+	Logger.Trace("Returning User Agent")
 	return getUserAgent("ci usage", callTree)
 }
 
 func getUserAgent(agent string, callTree string) string {
+	Logger.Trace("Obtaining parsed User Agent string")
 	return fmt.Sprintf("%s (%s; %s %s; %s)", getUserAgentBaseAndVersion(), agent, GOOS, GOARCH, callTree)
```

Comment:
Maybe pull this out into a var before returning it and log it too...might be useful to have the full built useragent in the future??

Reply:
Good point!

Reply:
In: 4dd0e60